tests/drivers/ipm/src/main.c

```diff
@@ -39,9 +41,9 @@ static struct ipm_console_sender_config_info sender_config = {
 	.bind_to = "ipm_dummy0",
 	.flags = SOURCE
 };
-DEVICE_INIT(ipm_console_send0, "ipm_send0", ipm_console_sender_init,
+DEVICE_AND_API_INIT(ipm_console_send0, "ipm_send0", ipm_console_sender_init,
 	    NULL, &sender_config,
-	    APPLICATION, INIT_PRIO_IPM_SEND);
+	    APPLICATION, INIT_PRIO_IPM_SEND, NULL);
```

Comment:
Needless "_INIT" and ", NULL"

Reply:
Can you pls elaborate?
modified DEVICE_INIT() to DEVICE_AND_API_INIT() as DEVICE_INIT() will tagged as deprecated